test/testsuite/lantest_io_monitor.c

```diff
@@ -0,0 +1,675 @@
+/*
+ * Copyright (c) 2025, Andy Lemin (andylemin@github.com)
+ * Credits; Based on work by Netatalk contributors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#ifdef __linux__
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <dirent.h>
+#include <pwd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <ctype.h>
+
+#include "lantest_io_monitor.h"
+
+#include <limits.h>  /* For PATH_MAX */
+
+/* Global variables for IO monitoring */
+int io_monitoring_enabled = 0;
+pid_t afpd_pid = 0;
+pid_t cnid_dbd_pid = 0;
+unsigned long long afpd_start_reads = 0, afpd_start_writes = 0;
+unsigned long long cnid_start_reads = 0, cnid_start_writes = 0;
+unsigned long long afpd_end_reads = 0, afpd_end_writes = 0;
+unsigned long long cnid_end_reads = 0, cnid_end_writes = 0;
```

Comment:
I have a preference for explicit integer types, like `uint64_t` over `unsigned long long`. The netatalk codebase has been gradually converted over to the explicit types over the years, but the testsuite code is still overall archaic since it was merged with netatalk proper fairly recently.